code/game/turfs/simulated/walls.dm

```diff
@@ -50,22 +50,19 @@
 			new /obj/effect/decal/remains/human(src)
 
 	else
-		if(!devastated)
-			new /obj/structure/girder(src)
-			if(mineral == "metal")
-				getFromPool(/obj/item/stack/sheet/metal, get_turf(src), 2)
-			else
-				var/M = text2path("/obj/item/stack/sheet/mineral/[mineral]")
-				new M(src)
-				new M(src)
+		if(mineral == "metal")

```

Comment:
This system is moronic and I don't like its extension.

It should be using material IDs and then getting sheet type from the material datum, not doing this weird path nonsense.